src/graph/impl/KokkosGraph_Distance2Color_impl.hpp

```diff
@@ -40,899 +40,1615 @@
+        non_const_1d_bool_view_t::HostMirror h_flags = Kokkos::create_mirror_view(d_flags);
+
+        // Initialize flags on host
+        for(int i = 0; i < 4; i++) h_flags(i) = false;
 
-    // Save out the number of phases and vertex colors
-    this->cp->set_vertex_colors( colors_out );
-    this->cp->set_num_phases( (double)iter );
+        // Deep copy initialized flags to device
+        Kokkos::deep_copy(d_flags, h_flags);
 
-  }   // color_graph_d2 (end)
+        functorVerifyDistance2Coloring vr(this->nv, xadj_, adj_, t_xadj_, t_adj_, vertex_colors_, d_flags, chunkSize_);
+        Kokkos::parallel_for("ValidateD2Coloring", my_exec_space(0, num_chunks), vr);
 
+        // Deep copy flags back to host memory
+        Kokkos::deep_copy(h_flags, d_flags);
 
+        validation_flags[ 0 ] = h_flags[ 0 ];
+        validation_flags[ 1 ] = h_flags[ 1 ];
+        validation_flags[ 2 ] = h_flags[ 2 ];
+        validation_flags[ 3 ] = h_flags[ 3 ];
 
-private:
+        output = !h_flags[ 0 ];
 
+        return output;
+    }      // verify_coloring (end)
 
-  // -----------------------------------------------------------------
-  //
-  // GraphColorD2::colorGreedy()
-  //
-  // -----------------------------------------------------------------
-  void colorGreedy(const_lno_row_view_t     xadj_,
-                   const_lno_nnz_view_t     adj_,
-                   const_clno_row_view_t    t_xadj_,
-                   const_clno_nnz_view_t    t_adj_,
-                   color_view_type          vertex_colors_,
-                   nnz_lno_temp_work_view_t current_vertexList_,
-                   nnz_lno_t                current_vertexListLength_)
-  {
-    //std::cout << ">>> WCMCLEN colorGreedy (KokkosGraph_Distance2Color_impl.hpp) <<<" << std::endl;
-    nnz_lno_t chunkSize_ = this->_chunkSize;
 
-    if (current_vertexListLength_ < 100*chunkSize_)
+
+    /**
+     * Generate a histogram of the distance-2 colors
+     *
+     * @param[out] histogram A Kokkos::View that will contain the histogram.
+     *                       Must be of size num_colors+1
+     *
+     * @return None
+     */
+    void compute_color_histogram(nnz_lno_temp_work_view_t& histogram)
     {
-      chunkSize_ = 1;
+        KokkosKernels::Impl::kk_get_histogram<typename HandleType::color_view_type, nnz_lno_temp_work_view_t, MyExecSpace>(
+          this->nv, this->gc_handle->get_vertex_colors(), histogram);
+
+        MyExecSpace::fence();
     }
 
-    functorGreedyColor gc(this->nv,
-                          xadj_,
-                          adj_,
-                          t_xadj_,
-                          t_adj_,
-                          vertex_colors_,
-                          current_vertexList_,
-                          current_vertexListLength_,
-                          chunkSize_
-                          );
-
-    Kokkos::parallel_for("KokkosGraph::Distance2Color::ColorGraphGreedy", my_exec_space(0, current_vertexListLength_ / chunkSize_ + 1), gc);
```

Comment:
Please make sure this is not accidental

Reply:
This is fine. It was changed to functorGreedyColorVB to be more consistent with other coloring naming schemes like functorGreedyColorVB_BIT, etc.